app/models/manageiq/providers/ansible_tower/shared/automation_manager/tower_api.rb

```diff
@@ -8,7 +8,9 @@ def create_in_provider(manager_id, params)
       tower_object = provider_collection(manager).create!(params)
 
       refresh(manager)
-      find_by!(:manager_id => manager.id, :manager_ref => tower_object.id)
+      target = find_by!(:manager_id => manager.id, :manager_ref => tower_object.id)
+      refresh(target)
```

Comment:
I'll try to fill in the details from what I know.

"We" want to make sure that the Tower Repos have done a `git pull` *before* we refresh them.  This allows the inventory collection to get the most up-to-date playbooks from the repos.

However, this is only implemented for targeted refreshes.  This means that while doing a targeted refresh of a Tower Repo, the refresh process will ask the repo to update (`git pull`), and waits until it knows that's done.  Then, it continue with the targeted refresh.

This is *not* done for full refreshes.  I guess, because it would mean that the full refresh would have to ask *all* of the repos to `git pull` and maybe it's too hard correlating all those events back together to mean "all repos are updated, continue full refresh".